### PR TITLE
SSCS-4869 interloc appeal received

### PR DESCRIPTION
**Pull request by arcotc**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java [hunk 1]
```diff
@@ -18,6 +24,23 @@ static final boolean isMandatoryLetter(NotificationEventType eventType) {
         return STRUCK_OUT.equals(eventType);
     }
 
+    static final boolean isFallbackLetterRequiredForSubscriptionType(NotificationWrapper wrapper, SubscriptionType subscriptionType, NotificationEventType eventType) {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java]: the final special word in the static method is not needed.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java [hunk 2]
```diff
@@ -18,6 +24,23 @@ static final boolean isMandatoryLetter(NotificationEventType eventType) {
...
+            && (APPELLANT.equals(subscriptionType)
+            || APPOINTEE.equals(subscriptionType)
+            || (REPRESENTATIVE.equals(subscriptionType) && null != wrapper.getNewSscsCaseData().getAppeal().getRep()))) {
+            result = true;
+        }
+
+        return result;
+    }
+
+    static final boolean isBundledLetter(NotificationEventType eventType) {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java]: same here

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java [hunk 3]
```diff
@@ -50,54 +51,47 @@ public void manageNotificationAndSubscription(NotificationWrapper notificationWr
...
             reminderService.createReminders(notificationWrapper);
         }
     }
 
-    private boolean isValidNotification(NotificationWrapper wrapper, Subscription
-            subscription, NotificationEventType notificationType) {
-        return (isMandatoryLetter(notificationType) || (subscription != null && subscription.doesCaseHaveSubscriptions()
-                && notificationValidService.isNotificationStillValidToSend(wrapper.getNewSscsCaseData().getHearings(), notificationType)
-                && notificationValidService.isHearingTypeValidToSendNotification(wrapper.getNewSscsCaseData(), notificationType)));
+    private boolean isValidNotification(NotificationWrapper wrapper, SubscriptionWithType
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java]: this if statement looks like is doing quite a few things. I'd make sure the different branches are well junit covered

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 4]
```diff
@@ -1,30 +1,41 @@
...
 import uk.gov.hmcts.reform.sscs.exception.NotificationServiceException;
 import uk.gov.hmcts.reform.sscs.factory.NotificationWrapper;
+import uk.gov.service.notify.NotificationClientException;
 
 @Service
 @Slf4j
 public class SendNotificationService {
     private static final String DIRECTION_TEXT = "Direction Text";
     static final String DM_STORE_USER_ID = "sscs";
+    public static final String NOTIFICATION_TYPE_LETTER = "Letter";
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: access can be private

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 5]
```diff
@@ -1,30 +1,41 @@
...
 
     @Value("${feature.bundled_letters_on}")
-    private Boolean bundledLettersOn;
+    Boolean bundledLettersOn;
+
+    @Value("${feature.letters_on}")
+    Boolean lettersOn;
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: access can be private

Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: Fixed in the next merge.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java (hunk 4 above)
Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: Fix in next push

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java (hunk 1 above)
Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java]: Fixed in next push.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java (hunk 2 above)
Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationValidService.java]: Fixed in next push.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java [hunk 6]
```diff
@@ -90,6 +112,47 @@ private void sendEmailNotification(NotificationWrapper wrapper, Subscription sub
...
+            NotificationHandler.SendNotification sendNotification = () -> {
+                Address addressToUse = getAddressToUseForLetter(wrapper);
+
+                sendLetterNotification(wrapper, notification, addressToUse);
+            };
+            notificationHandler.sendNotification(wrapper, notification.getLetterTemplate(), NOTIFICATION_TYPE_LETTER, sendNotification);
+        }
+    }
+
+    private void sendFallbackLetterNoficationToRepresentative(NotificationWrapper wrapper, Subscription subscription, Notification notification) {
```
[the remark this answers is not in the document]
Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/SendNotificationService.java]: Fixed in next push.

**Comment by arcotc** [comment]
This ticket has been superceded SSCS-4307, closing

**Pull request closed without merging**